pkg/apis/build/v1alpha1/image_validation.go

```diff
@@ -96,11 +97,19 @@ func (is *ImageSpec) validateCacheSize(ctx context.Context) *apis.FieldError {
 
 	if apis.IsInUpdate(ctx) {
 		original := apis.GetBaseline(ctx).(*Image)
-		if original.Spec.CacheSize != nil && is.CacheSize.Cmp(*original.Spec.CacheSize) < 0 {
-			return &apis.FieldError{
-				Message: "Field cannot be decreased",
-				Paths:   []string{"cacheSize"},
-				Details: fmt.Sprintf("current: %v, requested: %v", original.Spec.CacheSize, is.CacheSize),
+		if original.Spec.CacheSize != nil {
+			if ctx.Value(IsExpandable) == false && is.CacheSize.Cmp(*original.Spec.CacheSize) != 0 {
```

Comment:
I think checking first if the cache size is decreased and throw "Field cannot be decreased" then checking for expansion is more precise error handling